qutebrowser/config/config.py

```diff
@@ -731,28 +732,49 @@ def set_command(self, win_id, section_=None, option=None, value=None,
             print_ = True
         else:
             with self._handle_config_error():
-                if option.endswith('!') and option != '!' and value is None:
+                if option.endswith('!') and option != '!' and len(values) == 0:
+                    # Handle inversion as special cases of the cycle code path
                     option = option[:-1]
                     val = self.get(section_, option)
-                    layer = 'temp' if temp else 'conf'
                     if isinstance(val, bool):
-                        self.set(layer, section_, option, str(not val).lower())
+                        values = ('false', 'true')
                     else:
                         raise cmdexc.CommandError(
                             "set: Attempted inversion of non-boolean value.")
-                elif value is not None:
-                    layer = 'temp' if temp else 'conf'
-                    self.set(layer, section_, option, value)
-                else:
+                elif len(values) == 0:
                     raise cmdexc.CommandError("set: The following arguments "
                                               "are required: value")
+                elif not cycle and len(values) > 1:
+                    raise cmdexc.CommandError("set: Too many values provided")
+
+                layer = 'temp' if temp else 'conf'
+                self._set_next(layer, section_, option, values)
 
         if print_:
             with self._handle_config_error():
                 val = self.get(section_, option, transformed=False)
             message.info(win_id, "{} {} = {}".format(
                 section_, option, val), immediately=True)
 
+    def _set_next(self, layer, section_, option, values):
+        """Set the next value out of a list of values."""
+        if len(values) == 1:
+            # If we have only one value, just set it directly (avoid
+            # breaking stuff like aliases or other pseudo-settings)
+            self.set(layer, section_, option, values[0])
+        else:
+            # Otherwise, use the next valid value from values, or the
+            # first if the current value does not appear in the list
+            assert len(values) > 1
+            val = self.get(section_, option, transformed=False)
+            try:
+                idx = values.index(str(val))
+                idx = (idx + 1) % len(values)
+                value = values[idx]
+            except ValueError:
+                value = values[0]
```

Comment:
I originally imagined raising an error in this case, but now I'm thinking is better.